src/Appwrite/Platform/Workers/Messaging.php

```diff
@@ -676,8 +676,8 @@ private function buildSmsMessage(Document $message, Document $provider): SMS
     private function buildPushMessage(Document $message): Push
     {
         $to = $message['to'];
-        $title = $message['data']['title'];
-        $body = $message['data']['body'];
+        $title = $message['data']['title'] === '' ? null : $message['data']['title'];
+        $body = $message['data']['body'] === '' ? null : $message['data']['body'];
         $data = $message['data']['data'] ?? null;
```

Comment:
The `?? null` part only ever happens if the first part is already `null` so I don't see the reason to do `?? null` 

The null operator is used to set non-null value if value is null

Reply:
If the array key doesn't exist, accessing it's value would give an undefined key access warning without the null coalesce